### PR TITLE
Am tuannq new feature activity intent

### PR DESCRIPTION
**Pull request by tuannq201**

### CODE REVIEW

**Review by huyhn30 — CHANGES_REQUESTED** [review]
merge code

**Review by huyhn30 — CHANGES_REQUESTED** [review]

File: app/src/am_tuannq/com/example/asian1/issue1/CaculatorActivity.java [hunk 1]
```diff
@@ -18,5 +32,78 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                return;
+            }
+            mNumberOne = Double.parseDouble(mEdtNumberOne.getText().toString());
+            mNumberTwo = Double.parseDouble(mEdtNumberTwo.getText().toString());
+            mResult = mNumberOne + mNumberTwo;
+            mTvResult.setText((getString(R.string.text_result, mResult)));
+        });
+        mBtnSub.setOnClickListener(view -> {
+            mEdtNumberTwo.setError(null);
+            if (!validateForm()) {
```
Comment (huyhn30) [inline app/src/am_tuannq/com/example/asian1/issue1/CaculatorActivity.java:63]: if (!TextUtils.isEmpty(mEdtNumberOne.getText()) && !TextUtils.isEmpty(mEdtNumberTwo.getText())) {

} else {

}

File: app/src/am_tuannq/com/example/asian1/issue1/HomePageActivity.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    private void receiveDataFromLogin() {
+        Acount mAcount = new Acount();
+        mAcount = (Acount) getIntent().getSerializableExtra("acount");
+        try {
+            if (mAcount.getmEmail() != null) {
+                mTvReceiveData.setText(
+                        getString(
+                                R.string.receive_from_login_facebook,
```
Comment (huyhn30) [inline app/src/am_tuannq/com/example/asian1/issue1/HomePageActivity.java:36]: format code again

File: app/src/am_tuannq/com/example/asian1/issue1/HomePageActivity.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                                mAcount.getmPassword()));
+            }
+        } catch (NullPointerException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void receiveDataFromInformation() {
+        User mUser = new User();
+        mUser = (User) getIntent().getSerializableExtra("user");
```
Comment (huyhn30) [inline app/src/am_tuannq/com/example/asian1/issue1/HomePageActivity.java:47]: public static final String SEND_DATA_USER = "send_data_user";

mUser = getIntent().getSerializableExtra(SEND_DATA_USER);

File: app/src/am_tuannq/com/example/asian1/issue1/HomePageActivity.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        try {
+            if (mUser.getmName() != null) {
+                mTvReceiveData.setText(
+                        getString(
+                                R.string.receive_from_information,
+                                mUser.getmName(),
```
Comment (huyhn30) [inline app/src/am_tuannq/com/example/asian1/issue1/HomePageActivity.java:53]: format code again

File: app/src/am_tuannq/com/example/asian1/issue1/InformationActivity.java [hunk 5]
```diff
@@ -1,22 +1,76 @@
...
+        mEdtAddInformation = findViewById(R.id.edtAddInformation);
+        mBtnSendInformation = findViewById(R.id.btnSendInformation);
+    }
+
+    private void onClickButton() {
+        mBtnSendInformation.setOnClickListener(view -> {
+            if (validateForm()) {
+                Intent intent = new Intent(InformationActivity.this, HomePageActivity.class);
+                mUser = new User();
+                mUser.setmName(mEdtName.getText().toString().trim());
```
Comment (huyhn30) [inline app/src/am_tuannq/com/example/asian1/issue1/InformationActivity.java:48]: where data College and hobby ?

**Review by huyhn30 — CHANGES_REQUESTED** [review]
Check same bug again

File: app/src/am_tuannq/com/example/asian1/issue1/CaculatorActivity.java [hunk 6]
```diff
@@ -18,5 +33,76 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        mBtnPlus.setOnClickListener(view -> {
+            if (!validateForm()) {
+                return;
+            }
+            mNumberOne = Double.parseDouble(mEdtNumberOne.getText().toString());
+            mNumberTwo = Double.parseDouble(mEdtNumberTwo.getText().toString());
+            mResult = mNumberOne + mNumberTwo;
+            mTvResult.setText((getString(R.string.text_result, mResult)));
+        });
+        mBtnSub.setOnClickListener(view -> {
```
Comment (huyhn30) [inline app/src/am_tuannq/com/example/asian1/issue1/CaculatorActivity.java:61]: mBtnSub.setOnClickListener(view -> {
             if (validateForm()) {
                  mNumberOne = Double.parseDouble(mEdtNumberOne.getText().toString());
                  mNumberTwo = Double.parseDouble(mEdtNumberTwo.getText().toString());
                  mResult = mNumberOne - mNumberTwo;
                  mTvResult.setText((getString(R.string.text_result, mResult)));
             }
         });

File: app/src/am_tuannq/com/example/asian1/issue1/CaculatorActivity.java [hunk 7]
```diff
@@ -18,5 +33,76 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            if (mNumberTwo == 0) {
+                mEdtNumberTwo.setError(getString(R.string.error_number_divide_zero));
+                return;
+            }
+            mResult = mNumberOne / mNumberTwo;
+            mTvResult.setText((getString(R.string.text_result, mResult)));
+        });
+    }
+
+    private boolean validateForm() {
```
Comment (huyhn30) [inline app/src/am_tuannq/com/example/asian1/issue1/CaculatorActivity.java:95]:
```
    private boolean validateForm() {
        mEdtNumberTwo.setError(null);
        boolean isValidForm = true;
        if (TextUtils.isEmpty(mEdtNumberOne.getText())) {
            mEdtNumberOne.setError(getString(R.string.error_number_empty));
            isValidForm = false;
        }
        if (TextUtils.isEmpty(mEdtNumberTwo.getText())) {
            mEdtNumberTwo.setError(getString(R.string.error_number_empty));
            isValidForm = false;
        }
        return isValidForm;
    }
```

File: app/src/am_tuannq/com/example/asian1/issue1/HomePageActivity.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        receiveDataFromLogin();
+        receiveDataFromInformation();
+    }
+
+    private void initView() {
+        mTvReceiveData = findViewById(R.id.tvRecevieData);
+    }
+
+    private void receiveDataFromLogin() {
+        Acount mAcount = new Acount();
```
Comment (huyhn30) [inline app/src/am_tuannq/com/example/asian1/issue1/HomePageActivity.java:32]: Name Local Variable: acount
Name Global Variable: mAcount

Comment (huyhn30) [reply app/src/am_tuannq/com/example/asian1/issue1/HomePageActivity.java:32]: Acount acount = (Acount) getIntent().getSerializableExtra(SEND_DATA_ACOUNT);

**Review by huyhn30 — CHANGES_REQUESTED** [review]
check same bug

File: app/src/am_tuannq/com/example/asian1/issue1/HomePageActivity.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    private void initView() {
+        mTvReceiveData = findViewById(R.id.tvRecevieData);
+    }
+
+    private void receiveDataFromLogin() {
+        Acount acount = new Acount();
+        acount = (Acount) getIntent().getSerializableExtra(SEND_DATA_ACOUNT);
+        try {
```
Comment (taitqs) [inline app/src/am_tuannq/com/example/asian1/issue1/HomePageActivity.java:34]: Why use try catch here?

File: app/src/am_tuannq/com/example/asian1/issue1/HomePageActivity.java [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+            }
+        } catch (NullPointerException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void receiveDataFromInformation() {
+        User user = new User();
+        user = (User) getIntent().getSerializableExtra(SEND_DATA_USER);
+        try {
```
Comment (taitqs) [inline app/src/am_tuannq/com/example/asian1/issue1/HomePageActivity.java:47]: Why use try catch here?

File: app/src/am_tuannq/com/example/asian1/issue1/LoginActivity.java [hunk 11]
```diff
@@ -1,16 +1,66 @@
...
+        mEdtPassword = findViewById(R.id.edtPassword);
+        mBtnLogin = findViewById(R.id.btnLogin);
+    }
+
+    private void buttonOnClick() {
+        mBtnLogin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (validateForm()) {
+                    Intent intent = new Intent(LoginActivity.this, HomePageActivity.class);
```
Comment (huyhn30) [inline app/src/am_tuannq/com/example/asian1/issue1/LoginActivity.java:41]: Fix

File: app/src/am_tuannq/com/example/asian1/issue1/LoginActivity.java [hunk 12]
```diff
@@ -1,16 +1,66 @@
...
+    }
+
+    private void initView() {
+        mEdtEmail = findViewById(R.id.edtEmail);
+        mEdtPassword = findViewById(R.id.edtPassword);
+        mBtnLogin = findViewById(R.id.btnLogin);
+    }
+
+    private void buttonOnClick() {
+        mBtnLogin.setOnClickListener(new View.OnClickListener() {
```
Comment (huyhn30) [inline app/src/am_tuannq/com/example/asian1/issue1/LoginActivity.java:37]: use lambda

File: app/src/am_tuannq/com/example/asian1/issue1/HomePageActivity.java [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+        receiveDataFromLogin();
+        receiveDataFromInformation();
+    }
+
+    private void initView() {
+        mTvReceiveData = findViewById(R.id.tvRecevieData);
+    }
+
+    private void receiveDataFromLogin() {
+        Acount acount = new Acount();
```
Comment (huyhn30) [inline app/src/am_tuannq/com/example/asian1/issue1/HomePageActivity.java:32]: `Acount about = (Acount)getIntent().getSerializableExtra(SEND_DATA_ACOUNT);`

**Review by huyhn30 — CHANGES_REQUESTED** [review]
check again

**Review by huyhn30 — APPROVED** [review]

**Review by huyhn30 — APPROVED** [review]

**Pull request merged**